usr/src/uts/common/fs/zfs/dbuf.c

```diff
@@ -3284,13 +3284,14 @@ dbuf_write_children_ready(zio_t *zio, arc_buf_t *buf, void *vdb)
 	dmu_buf_impl_t *db = vdb;
 	dnode_t *dn;
 	blkptr_t *bp;
-	uint64_t i;
+	int i;
```

Comment:
Yeah, < 31 will work.  And I honestly thought lint would whine about integer overflow in that case, but that's expecting a bit much from lint, I admit.